addons/web/static/src/views/list/list_controller.js

```diff
@@ -509,10 +509,10 @@ export class ListController extends Component {
 
     get deleteConfirmationDialogProps() {
         const root = this.model.root;
-        const body =
-            root.isDomainSelected || root.selection.length > 1
-                ? this.env._t("Are you sure you want to delete these records?")
-                : this.env._t("Are you sure you want to delete this record?");
+        let body = deleteConfirmationMessage;
+        if (root.isDomainSelected || root.selection.length > 1) {
+            body = deleteConfirmationMessage.replace("record", "records");
+        }
```

Comment:
This will not work with translations. Besides, just making this one word plural might not be enough in some languages. I think we should either make records always plural, or have two distinct messages